troubleshooting/tutorials/remote-core-dump/README.md

```diff
@@ -0,0 +1,309 @@
+
+
+## Step 2: Modifying your deployment to capture a core file
+
+Follow the version of this step that fits your deployment mode. 
+
+- [Setup for Local Dev Machine or Instance](#setup-for-local-dev-machine-or-instance)
+- [Setup for ECS EC2](#setup-for-ecs-ec2)
+- [Setup for ECS Fargate](#setup-for-ecs-fargate)
+- [Setup for EKS EC2](#setup-for-eks-ec2)
+
+### Setup for Local Dev Machine or Instance
+
+Simply run the debug build of Fluent Bit with ulimit unlimited and with the `/cores` directory mounted onto your host:
+```
+docker run --ulimit core=-1 -v /somehostpath:/cores -v $(pwd):/fluent-bit/etc amazon/aws-for-fluent-bit:debug
+```
+
+You may need to customize the Docker run command to mount additional files or your AWS credentials. This is just an example. The command mounts the current working directory to `/fluent-bit/etc` which is the default directory for the main `fluent-bit.conf` config file- this assumes you have a config file in your current working directory.
+
+When the Fluent Bit debug image crashes, a core file should be outputted to `/somehostpath`. When that happens, proceed to [Step 4: Using GDB with a core file (crashed Fluent Bit)](#step-4-using-gdb-with-a-core-file-crashed-fluent-bit).  Alternatively, you can use `docker exec` to get a terminal into the container and follow [Step 3: Using GDB with a live Fluent Bit](#step-3-using-gdb-with-a-live-fluent-bit).
+
+### Setup for ECS EC2
+
+You will need to modify your existing Task Definition or CloudFormation or CDK or etc to include the following:
+
+1. A volume for the `/cores` directory in Fluent Bit that is mounted onto the host file system. This is necessary because when the core dump is produced, we need to save it somewhere, since containers are ephemeral. 
+2. Set `initProcessEnabled` in the task definition so that when Fluent Bit crashes or is killed, orphaned processes will be cleaned up gracefully. 
+3. Grant the `SYS_PTRACE` capability to the Fluent Bit container so that we can attach to it with a debugger. 
+4. Enable unlimited core ulimit. This ensures there is no limit on the size of the core file. 
+4. [Optional] Grant the Task Role ECS Exec permissions. This is necessary if you are debugging a Fluent Bit task that is still running and is frozen or misbehaving. You can use ECS Exec and `gdb` to attach the the live Fluent Bit process. 
+
+There is an example task definition in this directory named `ecs-ec2-task-def.json` which you can use as a reference. 
+
+#### Volume Mount for /cores directory
+
+Define a volume mount for the Fluent Bit container like this:
+
+```
+            "mountPoints": [{
+                "containerPath": "/cores",
+                "sourceVolume": "core-dump"
+            }],
+```
+
+Then, you can define the volume in the task definition to be a path on your EC2 host:
+
+```
+    "volumes": [{
+        "name": "core-dump",
+        "host" : {
+            "sourcePath" : "/var/fluentbit/cores"
+        },
+    }],
+```
```

Comment:
Is the volume mount also optional? For customers who are experiencing freezing rather than crashing, they may not need to set volumes up.